flink-formats/flink-json/src/main/java/org/apache/flink/formats/json/canal/CanalJsonDeserializationSchema.java

```diff
@@ -214,6 +215,7 @@ public void deserialize(@Nullable byte[] message, Collector<RowData> out) throws
         if (message == null || message.length == 0) {
             return;
         }
+        List<GenericRowData> genericRowDataList = new ArrayList<>();
```

Comment:
We can add a transient field to `CacalJsonDeserializationSchema` to avoid creating new list for every record, it's more efficient, WDYT?